rero_mef/marctojson/do_gnd_agent.py

```diff
@@ -45,17 +54,41 @@ def __init__(self, marc, logger=None, verbose=False, transform=True):
         if transform:
             self._transform()
 
+    def get_type(self):
+        """Get type of record.
+
+        Entitäten der GND (Satztypen) 075 $b TYPE $2 gndgen
+        - b Körperschaft
+        - f Konferenz
+        - g Geografikum
+        - n Person (nicht individualisiert)
+        - p Person (individualisiert)
+        - s Sachbegriff
+        - u Werk
+        """
+        for field_075 in self.marc.get_fields('075') or []:
+            if field_075['2'] == 'gndgen':
+                return RECORD_TYPES.get(field_075['b'])
+
     def _transform(self):
         """Call the transformation functions."""
-        if self.marc.get_fields('100') or \
-                self.marc.get_fields('110') or \
-                self.marc.get_fields('111'):
-            for func in dir(self):
-                if func.startswith('trans'):
-                    func = getattr(self, func)
-                    func()
+        record_type = self.get_type()
+        if record_type in ['bf:Person', 'bf:Organisation']:
+            if self.marc.get_fields('100') or \
+                    self.marc.get_fields('110') or \
+                    self.marc.get_fields('111'):
```

Comment:
from doc (https://pymarc.readthedocs.io/en/latest/#pymarc.record.Record.get_fields)
It should work,  isn't it ?
```python
if self.marc.get_fields('100', '110', '111'):
    ...
```